js/qmwp-bargraph-gauge-scatterplot-timeline_charts.js

```diff
@@ -1,742 +1,744 @@
-			if (numOfPoints > 400)
-			{
-				statisticalRelationshipText = "Significant";
-			}
-		}
-		else if (correlation > -0.1)
-		{
-			effectSizeText = "None";
-		}
-		else if (correlation > -0.3)
-		{
-			effectSizeText = "Weak Negative";
-			if (numOfPoints > 400)
-			{
-				statisticalRelationshipText = "Significant";
-			}
-		}
-		else if (correlation > -0.5)
-		{
-			effectSizeText = "Medium Negative";
-			if (numOfPoints > 45)
-			{
-				statisticalRelationshipText = "Significant";
-			}
-		}
-		else
-		{
-			effectSizeText = "Strong Negative";
-			if (numOfPoints > 17)
-			{
-				statisticalRelationshipText = "Significant";
-			}
-		}
-
-		jQuery('#statisticalRelationshipValue').text(statisticalRelationshipText);
-		jQuery('#effectSizeValue').text(effectSizeText);
-	}
+AnalyzeChart = function () {
+    var inputColor = '#42e212', outputColor = '#1a92c6', mixedColor = '#31B639', linearRegressionColor = '#bf8f1f';
+
+    var inputData = {
+        variableName: 'Data loading...',
+        unit: 'Data loading...',
+        timeSeries: [],
+        interpolant: function () {
+            return 0;
+        }
+    };
+    var outputData = {
+        variableName: 'Data loading...',
+        unit: 'Data loading...',
+        timeSeries: [],
+        interpolant: function () {
+            return 0;
+        }
+    };
+    var inputVariable, outputVariable;
+
+    var timelineChart, scatterplotChart, correlationGauge;
+
+    // Timeline settings
+    var tlSmoothGraph, tlGraphType; // Smoothgraph true = graphType spline
+    var tlEnableMarkers;
+    var tlEnableHorizontalGuides;
+
+    // Scatterplot settings
+    var spShowLinearRegression
+
+    var retrieveSettings = function () {
+        if (typeof(Storage) !== "undefined") {
+            tlEnableMarkers = (localStorage["tlEnableMarkers"] || "true") == "true" ? true : false;					// On by default
+            tlSmoothGraph = (localStorage["tlSmoothGraph"] || "true") == "true" ? true : false;						// On by default
+            tlEnableHorizontalGuides = (localStorage["tlEnableHorizontalGuides"] || "false") == "true" ? true : false;			// Off by default
+            spShowLinearRegression = (localStorage["spShowLinearRegression"] || "true") == "true" ? true : false;			// On by default
+
+            tlGraphType = tlSmoothGraph == true ? "spline" : "line"; // spline if smoothGraph = true
+        }
+    }
+
+    var setSettings = function (newSettings) {
+        if (typeof newSettings["tlSmoothGraph"] != "undefined") {
+            tlSmoothGraph = newSettings["tlSmoothGraph"];
+            tlGraphType = tlSmoothGraph == true ? "spline" : "line";
+            timelineChart.series[0].update({
+                type: tlGraphType
+            }, false);
+            timelineChart.series[1].update({
+                type: tlGraphType
+            }, false);
+            saveSetting("tlSmoothGraph", tlSmoothGraph);
+        }
+
+        if (typeof newSettings["tlEnableMarkers"] != "undefined") {
```

Comment:
Mixed double and single quotes.<br>['tlEnableMarkers'] is better written in dot notation.